### PR TITLE
[FEATURE] EC messages in the LoginView

### LINKED ISSUE #79: [FU] Authentication error handling and EC messages

### PR DESCRIPTION
**Pull request by mar-cas3**
* Disables the sign in button when there is no connection or the connection is low. 
* Re-enables the button once connection is restored
* Adds proper error management to the different cases identified.
* ⚠️ These cases were identified and implemented during testing in different conditions, including network speed and removing connections at different points during the process. However, the testing is no exhaustive, if new cases are found they should be reported as an issue and tagged as a bug 🐛
* Closes #79

https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/61592394/c00336f7-5e0a-4db3-8c88-d7e79513280e

### CODE REVIEW

File: FoodBookApp/UI/Views/Login/LoginView.swift [hunk 1]
```diff
@@ -64,15 +65,26 @@ struct LoginView: View {
...
+            .alert(viewModel.errorMsg, isPresented: $viewModel.showAlert) {
+                Button("OK", role: .cancel) { }
+            }
             
+            if(networkService.isUnavailable) {
+                Text("No connection, please make sure you have internet access before attempting to log-in")
+                    .foregroundStyle(.red)
+            }
+            
+            if(networkService.isLowConnection) {
```
Comment (Laurarestrepo03) [inline FoodBookApp/UI/Views/Login/LoginView.swift:84]: I have a question, shouldn't this be else if? (I might be wrong tho)

File: FoodBookApp/UI/Views/Login/LoginViewModel.swift [hunk 2]
```diff
@@ -6,13 +6,38 @@
...
+    func signInGoogle() async {
         let helper = SignInGoogleHelper()
-        let tokens = try await helper.signIn()
-        let authDataResult = try await AuthService.shared.signInWithGoogle(tokens: tokens)
+        
+        do {
+            let tokens = try await helper.signIn()
+            let authDataResult = try await AuthService.shared.signInWithGoogle(tokens: tokens)
+        } catch GIDSignInError.canceled {
+            errorMsg = "You canceled the sign-in flow. If you wish to sign in, press the button again and complete the authentication."
```
Comment (Laurarestrepo03) [inline FoodBookApp/UI/Views/Login/LoginViewModel.swift:26]: What does "cancel the sign-in flow" mean?

**Review by Laurarestrepo03 — CHANGES_REQUESTED** [review]

File: FoodBookApp/UI/Views/Login/LoginViewModel.swift (hunk 2 above)
Comment (mar-cas3) [reply FoodBookApp/UI/Views/Login/LoginViewModel.swift:26]: It's when they press the 'Cancel' button on the browser that pops up for the login!

File: FoodBookApp/UI/Views/Login/LoginView.swift (hunk 1 above)
Comment (mar-cas3) [reply FoodBookApp/UI/Views/Login/LoginView.swift:84]: I think the two conditions will never both be true at the same time. One is if there is no network connection at all which is set with the path.status of the network monitor and the other is directly a boolean of the path which is `isConstrained`. From the docs and testing, this flag is only true when you set your celullar options to Low Data Mode.

Comment (mar-cas3) [reply FoodBookApp/UI/Views/Login/LoginView.swift:84]: I can make the change so the second part of the message is shared among the two conditions, and leave as if's for now. I would still like to do better testing of these EC things to check edge cases and such so I would make the change as a bugfix.

**Pull request merged**